massa-pool-worker/src/tests/operation_pool_tests.rs

```diff
@@ -52,174 +42,164 @@ fn test_add_irrelevant_operation() {
-
-    let recv_keypair = KeyPair::generate();
-
-    let op = OperationType::Transaction {
-        recipient_address: Address::from_public_key(&recv_keypair.get_public_key()),
-        amount: Amount::default(),
-    };
-    let content = Operation {
-        fee: Amount::from_str(&fee.to_string()).unwrap(),
-        op,
-        expire_period,
-    };
-    Operation::new_verifiable(content, OperationSerializer::new(), &sender_keypair).unwrap()
-}
-
 /// TODO refactor old tests
 #[test]
 fn test_pool() {
-    let (execution_controller, execution_receiver) = MockExecutionController::new_with_receiver();
     let pool_config = PoolConfig::default();
-    let storage_base = Storage::create_root();
-    let operation_sender = broadcast::channel(pool_config.broadcast_operations_capacity).0;
-    let mut pool = OperationPool::init(
+    pool_test(
         pool_config,
-        &storage_base,
-        execution_controller,
-        PoolChannels { operation_sender },
-    );
-    // generate (id, transactions, range of validity) by threads
-    let mut thread_tx_lists = vec![Vec::new(); pool_config.thread_count as usize];
-    for i in 0..18 {
-        let fee = 40 + i;
-        let expire_period: u64 = 40 + i;
-        let start_period = expire_period.saturating_sub(pool_config.operation_validity_periods);
-        let op = get_transaction(expire_period, fee);
-        let id = op.id;
-
-        let mut ops = PreHashMap::default();
-        ops.insert(id, op.clone());
-        let mut storage = storage_base.clone_without_refs();
-        storage.store_operations(ops.values().cloned().collect());
-        pool.add_operations(storage);
-        //TODO: compare
-        // assert_eq!(storage.get_op_refs(), &Set::<OperationId>::default());
-
-        // duplicate
-        let mut storage = storage_base.clone_without_refs();
-        storage.store_operations(ops.values().cloned().collect());
-        pool.add_operations(storage);
-        //TODO: compare
-        //assert_eq!(storage.get_op_refs(), &ops.keys().copied().collect::<Set<OperationId>>());
-
-        let op_thread = op
-            .content_creator_address
-            .get_thread(pool_config.thread_count);
-        thread_tx_lists[op_thread as usize].push((op, start_period..=expire_period));
-    }
-    std::thread::spawn(move || loop {
-        match execution_receiver.recv_timeout(Duration::from_millis(100)) {
-            // forward on the operations
-            Ok(MockExecutionControllerMessage::UnexecutedOpsAmong {
-                ops, response_tx, ..
-            }) => {
-                response_tx.send(ops).unwrap();
+        |mut pool_manager, mut pool, execution_receiver, storage_base| {
+            // generate (id, transactions, range of validity) by threads
+            let mut thread_tx_lists = vec![Vec::new(); pool_config.thread_count as usize];
+
+            for i in 0..18 {
+                let expire_period: u64 = 40 + i;
+                let op = OpGenerator::default()
```

Comment:
Ideally this code could be handled by a helper method, as similar work is done in the other test functions. This is not so straight forward, as there are subtle differences between the tests that would make this doable, but in my view "clever for the sake of clever", and not of significant benefit in this context.